packages/fether-electron/src/main/index.js

```diff
@@ -66,7 +95,104 @@ app.on('will-quit', killParity);
 
 app.on('quit', () => {
   pino.info('Leaving Fether');
+  app.releaseSingleInstanceLock();
   killParity();
 });
 
+/**
+ * Security. Intercept new-window events (i.e. `window.open`) by
+ * overriding event.newGuest without using the supplied options tag
+ * to try to mitigate risk of an exploit re-enabling node integration
+ * despite being turned off in the configuration
+ * (i.e. `nodeIntegration: false`).
+ *
+ * References:
+ * - https://www.electronjs.org/blog/webview-fix
+ * - https://blog.scottlogic.com/2016/03/09/As-It-Stands-Electron-Security.html
+ */
+app.on('web-contents-created', (event, win) => {
+  win.on(
+    'new-window',
+    (event, newURL, frameName, disposition, options, additionalFeatures) => {
+      event.newGuest = null;
+
+      if (!options.webPreferences) {
+        options.webPreferences = {};
+      }
+
+      options.webPreferences.nodeIntegration = false;
+      options.webPreferences.nodeIntegrationInWorker = false;
+      options.webPreferences.webviewTag = false;
+      delete options.webPreferences.preload;
+    }
+  );
+});
+
+/**
+ * Security. Intercept and prevent new WebView (that may be used
+ * by an attacker to gain access to the file system) in addition
+ * to setting `webviewTag: false`
+ *
+ * Reference: https://www.electronjs.org/blog/webview-fix
+ */
+app.on('web-contents-created', (eventOuter, contents) => {
+  contents.on('will-attach-webview', (eventInner, webPreferences, params) => {
+    // Strip away preload scripts if unused or verify their location is legitimate
+    delete webPreferences.preload;
+    delete webPreferences.preloadURL;
+
+    // Disable Node.js integration
+    webPreferences.nodeIntegration = false;
+
+    // Verify URL being loaded
+    if (!params.src.startsWith('https://localhost.3000/')) {
+      eventOuter.preventDefault();
+      eventInner.preventDefault();
+    }
+  });
+
+  // Insecure TLS Validation - verify the application does not explicitly opt-out of TLS validation
+  // Reference: https://doyensec.com/resources/us-17-Carettoni-Electronegativity-A-Study-Of-Electron-Security-wp.pdf
+  app.on(
+    'certificate-error',
+    (event, webContents, url, error, certificate, callback) => {
+      if (url === 'https://localhost:3000/') {
```

Comment:
Why can't we just disable everything when there's a certificate error? Do we need to bypass certificates in dev mode? (https://localhost:3000/)

Reply:
I think we need certificates in both dev mode and production. I've pushed a commit showing proposed changes